src/circuit.rs

```diff
@@ -952,65 +952,68 @@ mod tests {
         value::{ValueCommitTrapdoor, ValueCommitment},
     };
 
+    fn generate_circuit_instance<R: RngCore>(mut rng: R) -> (Circuit, Instance) {
+        let (_, fvk, spent_note) = Note::dummy(&mut rng, None);
+
+        let sender_address = spent_note.recipient();
+        let nk = *fvk.nk();
+        let rivk = *fvk.rivk();
+        let nf_old = spent_note.nullifier(&fvk);
+        let ak: SpendValidatingKey = fvk.into();
+        let alpha = pallas::Scalar::random(&mut rng);
+        let rk = ak.randomize(&alpha);
+
+        let (_, _, output_note) = Note::dummy(&mut rng, Some(nf_old));
+        let cmx = output_note.commitment().into();
+
+        let value = spent_note.value() - output_note.value();
+        let rcv = ValueCommitTrapdoor::random(&mut rng);
```

Comment:
The change from `ValueCommitTrapdoor::zero` to `ValueCommitTrapdoor::random` here was only visible when comparing with whitespace changes disregarded; it seems fine to me but I want to check that it is intended.

Reply:
Yes it is intended; this snuck in because the test was implemented back when we had little crate structure, and that part was never updated. I noticed it while inspecting the serialized proof, because there was a run of all-zeroes due to having two dummy (zero-valued) notes and a zero `rcv` giving a `cv_net` of the identity.